### DISCUSSION TITLE
Отображения кастомного плэйсхолдера в DatePicker

### DISCUSSION THREAD

**Discussion by TarasovAleksei** [VKCOM/VKUI#2727] [Q&A] [answered]
Предложения по улучшению документации.
Добрый день.
В документации по компоненту DatePicker указаны такие атрибуты как dayPlaceholder, monthPlaceholder, yearPlaceholder. Однако при отображении компонента никакой реакции нет. Плэйсхолдер остается стандартным от input type={"date"}. С телефона вообще плэйсхолдер пустой. 
Каким образом можно добиться отображения кастомного плэйсхолдера? Есть ли возможность заменить саму иконку с календариком?

**Comment by SevereCloud** [comment, marked as answer]
### Отображение кастомного плэйсхолдера

Атрибуты `dayPlaceholder`, `monthPlaceholder`, `yearPlaceholder` предназначены только для десктопной версии с тремя [CustomSelect](https://vkcom.github.io/VKUI/#!/CustomSelect)-aми. Если вам необходим такой вид всегда, можно обернуть компонент AdaptivityProvider c `hasMouse`.

```tsx
<AdaptivityProvider hasMouse>
  <DatePicker ... />
</AdaptivityProvider>
```

### Иконка календаря

Иконка календаря появляется только в Google Chrome

<img width="947" alt="Снимок экрана 2022-06-20 в 14 05 42" src="https://user-images.githubusercontent.com/14944123/174588519-243fafb4-0453-4bcd-a047-70a7e47c285d.png">

Вы можете его отключить его с помощью css

```css
input[type="date"]::-webkit-calendar-picker-indicator {
  display:none;
}
```

В Firefox иконки календаря нет. Зато есть крестик

<img width="1012" alt="image" src="https://user-images.githubusercontent.com/14944123/174588674-4ea894b4-6a69-4f9b-8603-29529744df6b.png">

### DatePicker устарел

В `v5.0.0` мы откажемся от компонента DatePicker. Больше подробностей в #2697
